src/app/find-help/view-offer-accommodation/view-offer-accommodation.component.ts

```diff
@@ -39,9 +39,13 @@ export class ViewOfferAccommodationComponent implements OnInit {
   }
 
   navigateBack(): void {
-    this.router.navigate([this.router.url.replace(/\/[^/]+$/, '')], {
-      queryParams: this.storeUrlService.getParams(this.categoryRouteName),
-    });
+    if (window.history.state.redirectFromAccount) {
```

Comment:
Reading:
https://angular.io/api/router/NavigationBehaviorOptions#state
it seems that it's better to take this from Angular APIs rather than `history.state`. The latter is quite fragile - any regular navigation will reset `state` to null, even something like clicking on an internal anchor on the page. From the docs:
> Developer-defined state that can be passed to any navigation. Access this value through the [Navigation.extras](https://angular.io/api/router/Navigation#extras) object returned from the [Router.getCurrentNavigation() method](https://angular.io/api/router/Router#getcurrentnavigation) while a navigation is executing.
>
> After a navigation completes, the router writes an object containing this value together with a `navigationId` to `history.state`. The value is written when `location.go()` or `location.replaceState()` is called before activating this route.

So maybe take this from this `Navigation.extras` API?

The same applies to other usages of `history.state` here.

Reply:
Done